timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/AllNodesDisabledNamespacesUpdater.java

```diff
@@ -0,0 +1,271 @@
+    }
+
+    public ReenableNamespacesResponse reEnableOnAllNodes(ReenableNamespacesRequest request) {
+        Set<Namespace> namespaces = request.getNamespaces();
+
+        if (anyNodeIsUnreachable()) {
+            return ReEnableNamespaceResponses.unsuccessfulDueToPingFailure(namespaces);
+        }
+
+        List<SingleNodeUpdateResponse> responses = reEnableNamespacesOnAllNodes(request);
+        if (updateWasSuccessfulOnAllNodes(responses)) {
+            return ReenableNamespacesResponse.successful(SuccessfulReenableNamespacesResponse.of(true));
+        }
+
+        Set<Namespace> consistentlyLockedNamespaces = getConsistentFailures(responses);
+        if (!consistentlyLockedNamespaces.isEmpty()) {
+            return ReEnableNamespaceResponses.unsuccessfulDueToConsistentlyLockedNamespaces(
+                    consistentlyLockedNamespaces);
+        }
+
+        // TODO(gs): should only attempt to roll back locally?
+        UUID lockId = request.getLockId();
+        boolean rollbackSuccess = attemptDisableOnAllNodes(namespaces, lockId, responses.size());
+        if (rollbackSuccess) {
+            return ReEnableNamespaceResponses.unsuccessfulButRolledBack(namespaces, lockId);
+        }
+
+        return ReEnableNamespaceResponses.unsuccessfulAndRollBackFailed(namespaces, lockId);
+    }
+
+    // Ping
+    private boolean anyNodeIsUnreachable() {
+        return !isSuccessfulOnAllNodes(service -> new BooleanPaxosResponse(service.ping(authHeader)));
+    }
+
+    private boolean isSuccessfulOnAllNodes(Function<DisabledNamespacesUpdaterService, PaxosResponse> request) {
+        Collection<PaxosResponse> remoteResponses =
+                executeOnAllRemoteNodes(request).responses().values();
+        return remoteResponses.size() == remoteUpdaters.size()
+                && remoteResponses.stream().allMatch(PaxosResponse::isSuccessful);
+    }
+
+    // Disable
+    private boolean attemptDisableOnAllNodes(Set<Namespace> namespaces, UUID lockId, int expectedResponseCount) {
+        List<SingleNodeUpdateResponse> rollbackResponses = disableNamespacesOnAllNodes(namespaces, lockId);
+        return updateWasSuccessfulOnAllNodes(rollbackResponses, expectedResponseCount);
+    }
+
+    private List<SingleNodeUpdateResponse> disableNamespacesOnAllNodes(Set<Namespace> namespaces, UUID lockId) {
+        DisableNamespacesRequest request = DisableNamespacesRequest.of(namespaces, lockId);
+        Function<DisabledNamespacesUpdaterService, SingleNodeUpdateResponse> update =
+                service -> service.disable(authHeader, request);
+        Supplier<SingleNodeUpdateResponse> localUpdate = () -> localUpdater.disable(request);
+
+        return attemptOnAllNodes(namespaces, lockId, update, localUpdate);
+    }
+
+    // ReEnable
+    private boolean attemptReEnableOnAllNodes(Set<Namespace> namespaces, UUID lockId, int expectedResponseCount) {
```

Comment:
We should re enable locally regardless of what remotes say. It breaks the symmetry, but it is more defensive